soc/xtensa/intel_adsp/common/bootloader/boot_loader.c

```diff
@@ -6,6 +6,11 @@
  * Author: Liam Girdwood <liam.r.girdwood@linux.intel.com>
  */
 
+/* Older (GCC 4.2-based) XCC variants need a fixup file. */
+#if defined(__XCC__) && (__GNUC__ == 4)
+#include <toolchain/xcc_missing_defs.h>
+#endif
```

Comment:
That's a good point.  This was exposed because the bootloader is now including more Zephyr headers than it used to and ran afoul of this warning (for which we had a workaround, but done in a way that only works for "official" Zephyr .obj files and not side-built things like the bootloader).  Will move so it lives before the header changes.